spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java

```diff
@@ -237,4 +237,33 @@ public void testMiscellaneousExpressions() {
     assertThat(r1.get(8)).isEqualTo("glass"); // COALESCE
     assertThat(r1.get(9)).isEqualTo("working"); // IF CONDITION
   }
+
+  @Test
+  public void testBooleanExpressions() {
+    Dataset<Row> df =
+        spark
+            .read()
+            .format("bigquery")
+            .option("materializationDataset", testDataset.toString())
+            .load(TestConstants.SHAKESPEARE_TABLE);
+    df.createOrReplaceTempView("shakespeare");
+
+    List<Row> result =
+        spark
+            .sql(
+                "SELECT "
+                    + "word, "
+                    + "word LIKE '%las%' AS Contains, "
+                    + "word LIKE '%lass' AS Ends_With, "
+                    + "word LIKE 'gla%' AS Starts_With "
+                    + "FROM shakespeare "
```

Comment:
I was not able to trigger the rest of the boolean expressions since Spark would resolve those on its own. 

But I will try to change them in the next PR